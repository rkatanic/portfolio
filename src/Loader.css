.loader {
  width: 3rem;
  height: 0.75rem;
  margin-right: 1rem;
  position: relative;
}

.loader-square {
  width: 0.75rem;
  height: 0.75rem;
  background-color: var(--color-primary);
  position: absolute;
  animation: pulse 0.5s ease-in-out infinite alternate;
}

.square-1 {
  left: 0;
  transform-origin: 100% 50%;
}

.square-2 {
  left: 50%;
  transform: translateX(-50%) scale(1);
  animation-delay: 0.25s;
}

.square-3 {
  right: 0;
  animation-delay: 0.5s;
}

@keyframes pulse {
  to {
    transform: translateX(-50%) scale(0);
  }
}
