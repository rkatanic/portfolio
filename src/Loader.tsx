import "./Loader.css";

const Loader = (): JSX.Element => (
  <div className="loader">
    <div className="loader-square square-1"></div>
    <div className="loader-square square-2"></div>
    <div className="loader-square square-3"></div>
  </div>
);

export default Loader;
