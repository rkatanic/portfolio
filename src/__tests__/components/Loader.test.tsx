import { render } from "@testing-library/react";
import Loader from "../../components/Loader";

describe("Loader", (): void => {
  it("should render", (): void => {
    const { baseElement } = render(<Loader />);

    expect(baseElement).toMatchSnapshot();
  });
});
