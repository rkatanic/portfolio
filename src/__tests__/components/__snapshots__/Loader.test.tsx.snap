// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render 1`] = `
<body>
  <div>
    <div
      class="loader"
    >
      <div
        class="loader-square square-1"
      />
      <div
        class="loader-square square-2"
      />
      <div
        class="loader-square square-3"
      />
    </div>
  </div>
</body>
`;
